### PR TITLE
Add mask_tensor util

### PR DESCRIPTION
**Pull request by JackTemaki**
Utility to create a boolean tensor mask on the tensors device based on sequence lengths.

### CODE REVIEW

File: i6_models/util/mask.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+def mask_tensor(tensor: torch.Tensor, seq_len: torch.Tensor) -> torch.Tensor:
+    """
+    Create a "positive" mask for a tensor (boolean true means position is used)
+    on the same device as the tensor.
+
+    :param tensor: [B,T,....]
+    :param seq_len: [B]
+    :return: Mask of [B,T]
+    """
+    seq_len = seq_len.to(device=tensor.device)
```
Comment (Atticus1806) [inline i6_models/util/mask.py:13]: I am not sure if this case should be made within this function.

**Review by SimBe195 — APPROVED** [review]

File: i6_models/util/mask.py (hunk 1 above)
Comment (JackTemaki) [reply i6_models/util/mask.py:13]: The device handling? So completely removing tensor and having just a seq_len to mask function?

Comment (Atticus1806) [reply i6_models/util/mask.py:13]: you still need the tensor for the shape right? But I feel like this casting of the seq_len should be done beforehand (and at least in my setups usually already is done with the model call. So this might be redundant imho.

Comment (JackTemaki) [reply i6_models/util/mask.py:13]: You could do it without the tensor (using max(seq_len)), I think we can add this as a second function

Comment (JackTemaki) [reply i6_models/util/mask.py:13]: If seq_len is already on the device, this should have not more overhead than one integer comparison

Comment (Atticus1806) [reply i6_models/util/mask.py:13]: is this max then exportable? 
In general if the overhead is low I don't mind and we can leave it like this.

File: i6_models/util/mask.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import torch
+
+
+def mask_tensor(tensor: torch.Tensor, seq_len: torch.Tensor) -> torch.Tensor:
```
Comment (albertz) [inline i6_models/util/mask.py:4]: I'm not sure about the name. I find it a bit too generic. And also ambiguous. (Does it mask a tensor? No, actually, it returns a mask tensor. For what? Based on what?)

In TF, the corresponding function is called `tf.sequence_mask`. Why not just use the same name here? So all people who have used TF at some point in the past (almost everyone) knows what this is about.

[the remark this answers is not in the document]
Comment (JackTemaki) [reply i6_models/util/mask.py:4]: > What you actually want

What I want is to have it with a tensor and make sure the mask is on the same device as the tensor.  But maybe the majority disagrees, and I keep this for myself.

Comment (JackTemaki) [reply i6_models/util/mask.py:4]: tf.sequence_mask has a different signature, and we can also add such a function. This is just not what I wanted to have here.

Comment (JackTemaki) [reply i6_models/util/mask.py:4]: For the name, yes this can be more verbose if needed.

**Comment by JackTemaki** [comment]
Closing, as this specific signature seems not to be beneficial for people other than me.
